File: Day185_2025/Q185_Leetcode.py
class Solution:
    def kthCharacter(self, k: int, operations: list[int]) -> str:
        k -= 1  # convert to 0-based index
        ans = 0
        for i in range(60, -1, -1):  # supports up to 2^60 bits
            if (k >> i) & 1:
                if i < len(operations):
                    ans += operations[i]
        return chr(ord('a') + (ans % 26))

File: Day185_2025/Q185_Leetcode.java
class Q185_Leetcode {
    public char kthCharacter(long k, int[] operations) {
        k--;  // convert to 0-based indexing
        int ans = 0;
        for (int i = 63; i >= 0; i--) {
            if (((k >> i) & 1) == 1 && i < operations.length) {
                ans += operations[i];
            }
        }
        return (char) ('a' + (ans % 26));
    }
}
